### ISSUE TITLE
Are there plans to support Qwen-Image-Edit Inpainting LoRA training?

### ISSUE THREAD

**Issue by huan085128** [kohya-ss/musubi-tuner#532]
https://github.com/huggingface/diffusers/pull/12225

**Comment by kohya-ss** [comment]
The issue with Diffusers seems to be about inference. I don't think there's any need to train LoRA for inpainting.

Is there a case where we might need LoRA for inpainting?

**Comment by huan085128** [comment]
> The issue with Diffusers seems to be about inference. I don't think there's any need to train LoRA for inpainting.
> 
> Is there a case where we might need LoRA for inpainting?

I think it’s still useful in some cases—you can treat it like a FLUX fill model. For example, in object-removal: without a trained LoRA you have to prompt “remove this orange” to erase it, but once the LoRA is trained, just “remove this” does the trick.

**Comment by huan085128** [comment]
With an inpainting model, we don’t need to spell out the object—we just type in the editing target. That makes the prompt simpler, especially when the object itself is hard to describe.

**Comment by kohya-ss** [comment]
Thank you for the explanation. Wouldn't that be effective with masking during inference, not for training?
Masking during training calculates loss only in the masked parts. In other words, changes to parts other than the masked part will not affect the loss. This means that instead of training to keep the parts other than the mask constant, it learns to make arbitrary changes. I think this is not intended behavior.

**Comment by huan085128** [comment]
> Thank you for the explanation. Wouldn't that be effective with masking during inference, not for training? Masking during training calculates loss only in the masked parts. In other words, changes to parts other than the masked part will not affect the loss. This means that instead of training to keep the parts other than the mask constant, it learns to make arbitrary changes. I think this is not intended behavior.

Thank you. I have another question—when I use the Qwen edit LoRA trained with musubi-tuner on the inpainting pipeline, the results are pretty bad. I’m not sure why, since I already converted the LoRA to Diffusers format using convert_lora.py.

<table>
  
  <tr>
    <td align="center">Source</td>
    <td align="center">Mask</td>
    <td align="center">qwen edit inpainting</td>
    <td align="center">qwen edit inpainting + lora</td>
  </tr>
  
  <tr>
    <td>
      <img src="https://i.imgur.com/B7cfqWA.jpeg" width="200">
    </td>
    <td>
      <img src="https://i.imgur.com/SvWksjq.png" width="200">
    </td>
    <td>
      <img src="https://i.imgur.com/1MLn7fL.png" width="200">
    </td>
    <td>
      <img src="https://i.imgur.com/4y0WQdZ.jpeg" width="200">
    </td>
  </tr>
</table>

inference code:
```python
import torch
from diffusers import QwenImageEditInpaintPipeline
from diffusers.utils import load_image
import os

pipe = QwenImageEditInpaintPipeline.from_pretrained("Qwen-Image-Edit", torch_dtype=torch.bfloat16)
pipe.load_lora_weights("output/0908_qwen_edit_inpainting-000009_diffusers.safetensors")

pipe.to("cuda:5")
prompt = "add a banana"
negative_prompt = ""
source = load_image("glasses-1285273_1280 (1).jpg")
mask = load_image("ComfyUI_temp_zqkkx_00002_.png")

image = pipe(
    prompt=prompt,
    negative_prompt=negative_prompt,
    image=source,
    mask_image=mask,
    strength=1.0,
    num_inference_steps=35,
    true_cfg_scale=4.0,
    generator=torch.Generator(device="cuda").manual_seed(41412)
).images[0]
image.save(f"qwen_inpainting_6.png")
```

To add some details: this LoRA model was trained on a dozen or so before-and-after data pairs, with the label “add some banana on the wooden board.” When I use the original pipeline, the results look fine.
